addons/mrp/models/mrp_workorder.py

```diff
@@ -231,32 +320,85 @@ def record_production(self):
         # Transfer quantities from temporary to final move line or make them final
         self._update_raw_moves()
 
+        # Record the final lot and quantity produced to suggest on next workorders
+        if self.product_tracking != 'none' and self.final_lot_id:
+            self._update_final_lot_line()
+
         # Update workorder quantity produced
         self.qty_produced += self.qty_producing
 
-        if self.final_lot_id:
-            self.final_lot_id.use_next_on_work_order_id = self.next_work_order_id
-            self.final_lot_id = False
-
+        if self.next_work_order_id and self.production_id.product_id.tracking != 'none' and not self.next_work_order_id.final_lot_id:
+            self.next_work_order_id._assign_default_final_lot_id(self.final_lot_line_ids)
+            self.next_work_order_id._update_wo_lines()
```

Comment:
add a comment: as we may have changed the quantity to produce on the next workorder, make sure to update its wokorder lines